enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import { useDisclosure } from "@mantine/hooks";
+import { useState } from "react";
+
+import type { InteractiveQuestionProps } from "embedding-sdk/components/public/InteractiveQuestion";
+import { InteractiveQuestion } from "embedding-sdk/components/public/InteractiveQuestion";
+import { Box, Group, Overlay, Paper, Tabs } from "metabase/ui";
+
+import { useInteractiveQuestionContext } from "../InteractiveQuestion/context";
+
+const QuestionEditorInner = () => {
+  const { queryResults, runQuestion } = useInteractiveQuestionContext();
+
+  const [activeTab, setActiveTab] = useState<
+    "notebook" | "visualization" | (string & unknown) | null
+  >("notebook");
+  const [isSaveFormOpen, { open: openSaveForm, close: closeSaveForm }] =
+    useDisclosure(false);
+
+  const onOpenVisualizationTab = async () => {
+    setActiveTab("visualization");
+    await runQuestion();
+  };
+
+  return (
+    <Box w="100%" h="100%">
+      <Tabs
+        value={activeTab}
+        onTabChange={setActiveTab}
+        defaultValue="notebook"
+      >
+        <Group position="apart">
+          <Group>
+            <Tabs.Tab value="notebook">Notebook</Tabs.Tab>
+            {queryResults ? (
```

Comment:
we only show the option to open the `Visualization` table when a query has already been run. This basically gives the user one path to create a question, click `Visualize`, and then go from there

Reply:
Tested. This works great!